### PR TITLE
Change Path Search Priority to favor "extra"

### LINKED ISSUE #289: extra folder should have search priority over user added search paths
Hi, in cyclone we have a hilbert~ object, that I'm trying to load only as [cyclone/hilbert~], so just "[hilbert~]" would still call the abstraction that resides in "extra".

But that never happens as it seems externals have priority in the search path before the extra folder.

So the only way I can load pd vanilla's hilbert~ is if I don't have cyclone in Pd's path or in [declare], which is bad...

this could be fixed if extra had priority in the search, which I think it'd be common sense, as Pd Vanilla objects actually do have priorities over externals in general.

cheers

### PR DESCRIPTION
**Pull request by porres**
commit https://github.com/pure-data/pure-data/commit/d5296b3432a13a72f71bdf090e40a552b52ef92b partially closes #289 
fixes the priority as stated by @umlaeute in https://github.com/pure-data/pure-data/issues/289#issuecomment-357187775 - but user added paths are still searched first.

### CODE REVIEW

**Comment by porres** [comment]
> but user added paths are still searched first.

Actually, no... so it completely closes https://github.com/pure-data/pure-data/issues/289

File: src/s_path.c [hunk 1]
```diff
@@ -237,10 +237,12 @@ void sys_setextrapath(const char *p)
 {
     char pathbuf[MAXPDSTRING];
     namelist_free(STUFF->st_staticpath);
+    /* add built-in "extra" path first so its checked first */
+    STUFF->st_staticpath = namelist_append(0, p, 0);
```
Comment (danomatika) [inline src/s_path.c]: Wouldn't this line suffice? I think you are appending multiple copies with each append...

**Comment by porres** [comment]
I'm closing this as it got bluntly rejected, and I'll just work it out in the documentation ;)

**Comment by danomatika** [comment]
Often it helps to ask before doing the work if you are not sure. :)

**Comment by danomatika** [comment]
AKA don't burn yourself out spending energy if it may not make sense. Do a PR when you have a *reasonable* assumption it would be a valid proposal. :)

**Comment by porres** [comment]
> AKA don't burn yourself out spending energy if it may not make sense.

Sure, I see, but it's cool, I didn't mean it in a bad way, it was a bit dramatic for humorous purposes. I actually don't mind hitting a wall. I'm at this phase where I'm really enjoying trying to figure out Pd's source code and then edit it to see if it does what I think it should. So I'm always glad I learn something ;)

**Comment by danomatika** [comment]
I didn't mean it in a bad way either, I just don't want to lose a valuable contributor if you get burned out. :) We've been through that before and I also have to be much more selective in my time now that I have a family, so I'm more sensitive to it.

**Comment by danomatika** [comment]
Also, in regards to worrying about changing a precedent (even when it makes sense): https://xkcd.com/1172/

**Comment by porres** [comment]
Thanks, that really made me laugh hard :)

**Pull request closed without merging**